contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,301 @@
+    }
+
+    /**
+     * @notice Returns the amount of backing WETH the strategy holds
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_asset == address(weth), "Unsupported asset");
+
+        // Figure out how much wOETH is worth at the time.
+        // Always uses the last stored oracle price.
+        // Call updateWOETHOraclePrice manually to pull in latest yields.
+
+        // NOTE: If the contract has been deployed but the call to
+        // `updateWOETHOraclePrice()` has never been made, then this
+        // will return zero. It should be fine because the strategy
+        // should update the price whenever a deposit/withdraw happens.
+
+        // If `updateWOETHOraclePrice()` hasn't been called in a while,
+        // the strategy will underreport its holdings but never overreport it.
+
+        balance =
+            (bridgedWOETH.balanceOf(address(this)) * lastOraclePrice) /
+            1 ether;
+    }
+
+    /**
+     * @notice Check if an asset is supported.
+     * @param _asset    Address of the asset
+     * @return bool     Whether asset is supported
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        // Strategist deposits bridged wOETH but the contract only
+        // reports the balance in WETH. As far as Vault is concerned,
+        // it isn't aware of bridged wOETH token
+        return _asset == address(weth);
+    }
+
+    /***************************************
+               Overridden methods
+    ****************************************/
+
+    /**
+     * @notice deposit() function not used for this strategy
+     */
+    function deposit(address, uint256) public override onlyVault nonReentrant {
+        // Use depositBridgedWOETH() instead
+        require(false, "Deposit disabled");
+    }
+
+    /**
+     * @notice depositAll() function not used for this strategy
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        // Use depositBridgedWOETH() instead
+        require(false, "Deposit disabled");
+    }
+
+    /**
+     * @notice withdraw() function not used for this strategy
+     */
+    function withdraw(
+        // solhint-disable-next-line no-unused-vars
+        address _recipient,
+        // solhint-disable-next-line no-unused-vars
+        address _asset,
+        // solhint-disable-next-line no-unused-vars
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(false, "Withdrawal disabled");
+    }
+
+    /**
+     * @notice withdrawAll() function not used for this strategy
+     */
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        require(false, "Withdrawal disabled");
```

Comment:
I think we probably want to change this to `return false`, since we tend to want withdrawAll to succeed on the vault pulling as much funds as possible (from other strategies)

Reply:
`withdrawAll` signature doesn't have a return value. So, I'm just removing the `require` and leaving this as a blank function